src/main/webapp/js/quiz.js

```diff
@@ -0,0 +1,153 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+(window.onload = function(){
+  function buildQuiz(){
+    const output = [];
+    myQuestions.forEach(
+      (currentQuestion, questionNumber) => {
+        const answers = [];
+        for(letter in currentQuestion.answers){
+          answers.push(
+            `<label>
+              <input type="radio" name="question${questionNumber}" value="${letter}">
+              ${letter} :
+              ${currentQuestion.answers[letter]}
+            </label>`
+          );
+        }
+        output.push(
+          `<div class="slide">
+            <div class="question"> ${currentQuestion.question} </div>
+            <div class="answers"> ${answers.join("")} </div>
+          </div>`
+        );
+      }
+    );
+    quizContainer.innerHTML = output.join('');
+  }
+
+  function showResults(){
+    const answerContainers = quizContainer.querySelectorAll('.answers');
+    myQuestions.forEach( (currentQuestion, questionNumber) => {
+      const answerContainer = answerContainers[questionNumber];
+      const selector = `input[name=question${questionNumber}]:checked`;
+      const userAnswer = (answerContainer.querySelector(selector) || {}).value;
+        answerContainers[questionNumber].style.color = 'lightgreen';
+    });
+    resultsContainer.innerHTML = `Done! Click next for your product recommendations!`;
+  }
+
+  function showSlide(n) {
+    slides[currentSlide].classList.remove('active-slide');
+    slides[n].classList.add('active-slide');
+    currentSlide = n;
+    if(currentSlide === 0){
+      previousButton.style.display = 'none';
+    }
+    else{
+      previousButton.style.display = 'inline-block';
+    }
+    if(currentSlide === slides.length-1){
+      nextButton.style.display = 'none';
+      submitButton.style.display = 'inline-block';
+    }
+    else{
+      nextButton.style.display = 'inline-block';
+      submitButton.style.display = 'none';
+    }
+  }
+
+  function showNextSlide() {
+    showSlide(currentSlide + 1);
+  }
+
+  function showPreviousSlide() {
+    showSlide(currentSlide - 1);
+  }
+  const quizContainer = document.getElementById('quiz');
+  const resultsContainer = document.getElementById('results');
+  const submitButton = document.getElementById('submit');  
+  const myQuestions = [
+  {
+    question: "What is your perferred color family?",
+    answers: {
+      a: "Warm colors: reds, oranges, yellows",
+      b: "Cool colors: blues, greens, purples",
+      c: "Neutrals: browns, black/grey, white",
```

Comment:
Does this mean that we need to have these types of categories for each product?

If not, then how are we going to use the information of this answer?